### PR TITLE
feat: add namespace to schemas

### PR DESCRIPTION
**Pull request by rgraber**
**Description:**
Add namespace to event schemas. This will allow us to use the TopicRecordName strategy when registering schemas (otherwise, all the record names would just be "CloudEvent" and there would be name conflicts). This should be able to be deployed the producer and consumer in any order.

This will make the schema for a topic look like
```
{
 "doc": "Avro Event Format for CloudEvents created with openedx_events/schema",
  "fields": [<no change>],
  "name": "CloudEvent",
  "namespace": "org.openedx.content_authoring.course.catalog_info.changed.v1",
  "type": "record"
}
```

**ISSUE:**
https://github.com/openedx/openedx-events/issues/78

**Reviewers:**
- [ ] tag reviewer
- [ ] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

**Review by mariajgrimaldi — APPROVED** [review]
LGTM :). Can you add an example of how this would look like? So it's documented

**Comment by rgraber** [comment]
Ack, I should have left this in draft while I was still testing, sorry! I'm just making sure it can be deployed independently on producers and consumers without issue

**Comment by robrap** [comment]
@rgraber: Does this break CloudEvent requirements? See https://github.com/cloudevents/spec/blob/v1.0.1/avro-format.md#24-definition. If it does, we'll need determine if we move forward anyway and just note this somewhere. Note that @ormsbee selected CloudEvent because it seemed useful (he can say more), at the same time, I don't think it matters for anything at this point. Depending on your answer, we can discuss with Dave.

**Comment by rgraber** [comment]
> @rgraber: Does this break CloudEvent requirements? See https://github.com/cloudevents/spec/blob/v1.0.1/avro-format.md#24-definition. If it does, we'll need determine if we move forward anyway and just note this somewhere. Note that @ormsbee selected CloudEvent because it seemed useful (he can say more), at the same time, I don't think it matters for anything at this point. Depending on your answer, we can discuss with Dave.

It's unclear to me if that exact namespace is required. If anything, this would be a lateral move since we didn't have any namespace there to begin with so I don't know if we need to answer that now.

**Comment by rgraber** [comment]
@mariajgrimaldi this is ready for review, sorry about the mixup earlier

File: openedx_events/event_bus/avro/tests/test_schema.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ def test_simple_schema_generation(self):
             "type": "record",
             "name": "CloudEvent",
             "doc": "Avro Event Format for CloudEvents created with openedx_events/schema",
+            'namespace': 'simple.signal',
```
Comment (robrap) [inline openedx_events/event_bus/avro/tests/test_schema.py:37]: Nit: All this repetition makes me think a helper for generating this data might clean-up tests.

**Review by robrap — APPROVED** [review]
Looks fine, but please update the commit comment (and maybe PR description) with more details around why this is being added, and potentially pointing to docs regarding its need. Thank you!

**Comment by robrap** [comment]
@rgraber: That's fine. It was more about the commit message being empty. If there isn't a good doc to point to, forget about that.

**Comment by robrap** [comment]
@rgraber: In Slack, but possibly not here, you had written:
> It’s something we should do regardless of how we go about putting multiple events on one topic because it distinguishes the names of schemas between different events (since they’re all called “CloudEvent”).

Since we probably aren't using TopicRecordName strategy (please confirm), then shouldn't we either close this PR, or better document why we are doing this? I think your Slack comment is why you think it is important, right? Thanks.

**Comment by rgraber** [comment]
> @rgraber: In Slack, but possibly not here, you had written:
> 
> > It’s something we should do regardless of how we go about putting multiple events on one topic because it distinguishes the names of schemas between different events (since they’re all called “CloudEvent”).
> 
> Since we probably aren't using TopicRecordName strategy (please confirm), then shouldn't we either close this PR, or better document why we are doing this? I think your Slack comment is why you think it is important, right? Thanks.

Ah, I see looking back in Slack that I was not very clear.
We probably will still use the TopicRecordName strategy but we will be setting it on the client rather than the broker. This we can do regardless of cluster type. Thus, we will still need this PR to ensure that the schema for each signal type on a topic gets registered separately.
I believe we will need this PR even if we change our minds about using the TopicRecordName strategy vs avro unions because Avro unions refer to other schemas by name, so we'll need the names to be distinct.

**Comment by rgraber** [comment]
Am I still good to merge?

**Comment by rgraber** [comment]
I'm going to go ahead and merge so we can unblock the xblock work. Will put the rationale in the extended commit message.

**Pull request merged**